bolt.rb

```diff
@@ -69,10 +69,25 @@ def list
   end
 end
 
+def getshell(target)
+  # Bolt's inventory defines a shell as a feature. Some transports provide
+  # default features as well. Use these to determine the login shell.
+  if target.features.include?('powershell')
+    'powershell'
+  elsif target.features.include?('bash')
+    'posixshell'
+  elsif target.transport == 'winrm'
+    'powershell'
+  elsif target.transport == 'ssh' || target.transport.nil?
+    'posixshell'
+  end
+end
+
 class Target < Wash::Entry
   label 'target'
   parent_of VOLUMEFS
   state :target
+  attributes :loginshell
```

Comment:
Still need to decide on the final form of this attribute @ekinanp. Want to propose an alternative?

Reply:
Updated. I'll work on tests tomorrow.